src/main/java/io/sharedstreets/data/SharedStreetsOSMMetadata.java

```diff
@@ -160,8 +160,17 @@ public Way.ROAD_CLASS getRoadClass() {
         for(WaySectionMetadata waySection : this.waySections) {
             if(roadClass == null || roadClass == waySection.roadClass)
                 roadClass = waySection.roadClass;
-            else {
-                roadClass = Way.ROAD_CLASS.ClassOther;
+
+            // if mixed segment make segment class equal lowest class
+            // per https://github.com/sharedstreets/sharedstreets-ref-system/issues/20#issuecomment-378079937
+            //               [excluded road]
+            //                        |
+            //    [highway=primary]   |  [highway=secondary]
+            //    ====================*=====================
+            //               [roadClass=secondary]
+
+            else if(waySection.roadClass.getValue() > roadClass.getValue()){
+                roadClass = waySection.roadClass;
```

Comment:
So, higher road class number is the "lowest grade" of road, and here we are ensuring that if a segment contains multiple road classes, we use the "lowest grade"?